gigantum/gigantum-client
# add ability to provide ssl cert/key to the client

## Pull Request for Issue #1642

**Describe changes introduced by this PR (Bugs fixed, features added, docs updated, etc.)**:
This PR modifies some of the build and start up processes to support providing SSL certificates to the Client, primarily for when running as a multi-tenant client as a service (typically on an internal corporate network).

If a cert/key pair is placed in `~/gigantum/certificates/ssl/` the files will be copied into the proper location at start up and the proxy configured to use them.

Some additional clean up to gtm was done as well.



**Please check if the PR fulfills these requirements**

  - PR is set to merge to integration
  - PR title is descriptive, as this becomes the commit message on squash and merge.
  - Any related issues are added to the extended commit message during squash and merge.
  - Tests have been added (for bug fixes/features)
  - Docs have been added/updated (for bug fixes/features)

**QA Test Steps and Exit Criteria**:
- You should be able to place a valid cert/key pair into `~/gigantum/certificates/ssl/` and the server will run with HTTPS enabled
